server/endpoint/status/endpoint.go

```diff
@@ -83,6 +85,19 @@ func (e Endpoint) Endpoint() kitendpoint.Endpoint {
 
 		request := r.(Request)
 
+		cr, err := e.k8sClient.G8sClient().ApplicationV1alpha1().Apps(request.AppNamespace).Get(ctx, request.AppName, metav1.GetOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		if !key.InCluster(*cr) {
+			if request.Token == "" {
+				return nil, microerror.Maskf(wrongTokenError, "token empty")
+			} else if cr.GetResourceVersion() != request.Token {
+				return nil, microerror.Maskf(wrongTokenError, "incorrect token %#q ", request.Token)
```

Comment:
Do we need to worry about race conditions here?

What happens if the resource version changes because of another update to the app CR? Should we just reject the request and log instead or erroring?

Something else may have updated the app CR and changed

Reply:
You're right about that. Any changes to metadata lead to the rejection of status update. 
What do you think about using `generation` instead of `resourceVersion`in metadata, since we only change the generation when the spec is changed.

Reply:
Generation is a lot easier to brute force so that is quite a big drawback.

This is a tricky issue. Let's take a step back and see what is the best way to secure the endpoint.

Reply:
I thought about the process, and any `resourceVersion` changes before the status update are very unlikely because whenever app CR has been changed, it will update chart CRs with the latest resourceVersion from app CRs. 

The only thing that I don't like is that this would trigger cascading reconciliation into the app-operator and chart-operator *until* chart-operator sees no difference in chart CRs' status.

But this may be the reasonable cost that we have to pay to update the status immediately. WDYT?